scheduler/src/cook/mesos/api.clj

```diff
@@ -668,55 +668,59 @@
     group))
 
 
-(defn get-executor-states-impl
-  "Builds an indexed version of all executor states on the specified slave. Has no cache; takes
+(defn get-executor-id->directory-impl
+  "Builds an indexed version of all executor-id to directory on the specified agent. Has no cache; takes
    100-500ms to run."
   [framework-id hostname]
   (let [timeout-millis (* 5 1000)
         ;; Throw SocketTimeoutException or ConnectionTimeoutException when timeout
         slave-state (:body (http/get (str "http://" hostname ":5051/state.json")
-                                     {:socket-timeout timeout-millis
+                                     {:as :json-string-keys
                                       :conn-timeout timeout-millis
-                                      :as :json-string-keys
+                                      :socket-timeout timeout-millis
                                       :spnego-auth true}))
-        framework-executors (for [framework (reduce into
-                                                    []
-                                                    [(get slave-state "frameworks")
-                                                     (get slave-state "completed_frameworks")])
-                                  :when (= framework-id (get framework "id"))
-                                  e (reduce into
-                                            []
-                                            [(get framework "executors")
-                                             (get framework "completed_executors")])]
-                              e)]
-    (map-from-vals #(get % "id") framework-executors)))
+        framework-filter (fn framework-filter [{:strs [id] :as framework}] (when (= framework-id id) framework))
+        ;; we expect only one framework to match
```

Comment:
Changed to `there should be at most one framework entry for a given framework-id`